### DISCUSSION TITLE
Documenting the Teal API?

### DISCUSSION THREAD

**Discussion by daelvn** [teal-language/tl#378] [General]
While trying to use the Teal API, I found myself having to resource to inspecting the types reported by `tl check`, or reading the source code directly, because there seems to be no API documentation. Should it be documented? If so, should the documentation be written manually, in LDoc, or with an alternative project?

**Comment by hishamhm** [comment]
I think it's probably too early in these 0.x days, because the APIs are changing rapidly and the boundaries of what are public and private APIs are not well-established. Anything using `require("tl")` and using the functions for anything other than `tl.loader()` should be considered experimental for now.

**Reply by hishamhm** [reply]
When the time comes, I will document it LDoc-style!

**Reply by modawang** [reply]
Is there a document connection for TealAPI?

**Reply by hishamhm** [reply]
No, no formal documentation yet.
